### PR TITLE
5

### PR DESCRIPTION
**Pull request by charlie-storer-vensta**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -59,4 +59,45 @@ describe("Returns with the correct JSON endpoint", () => {
...
+          expect(input).toMatchObject(expectedOutput);
+        });
+      });
+  });
+  test("/api/reviews", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then((response) => {
+        response.body.reviews.forEach((review) => {
```
Comment (leekli) [inline __tests__/api.test.js]: I would recommend an assertion above the .foreach to check the length of the array, to check /api/reviews gives you the exact number of reviews you are expecting to see

File: __tests__/api.test.js [hunk 2]
```diff
@@ -59,4 +59,45 @@ describe("Returns with the correct JSON endpoint", () => {
...
+          expect(typeof review.created_at).toBe("string");
+          expect(typeof review.votes).toBe("number");
+          expect(typeof review.comment_count).toBe("string");
+          expect(typeof review.owner).toBe("string");
+          expect(typeof review.category).toBe("string");
+          expect(typeof review.review_img_url).toBe("string");
+          expect(typeof review.designer).toBe("string");
+          const input = Object.keys(
+            endpoint["GET /api/reviews"].exampleResponse
+          );
```
Comment (leekli) [inline __tests__/api.test.js]: No need to test against the endpoints.json file, your earlier tests on the /api endpoint should hopefully have sorted that for you

File: api/app.controller.js [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
+      response.status(200).send({ review: review });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+exports.getReviews = (request, response, next) => {
+  getReview()
+    .then((review) => {
+      response.status(200).send({ reviews: review });
```
Comment (leekli) [inline api/app.controller.js:33]: Good use of the `reviews` key name to represent sending all reviews back

File: api/app.models.js [hunk 4]
```diff
@@ -6,3 +6,28 @@ exports.getCategory = () => {
...
+exports.getReview = (sort_by = "created_at") => {
+  const validSortQueries = [
+    "created_at",
+    "votes",
+    "comment_count",
+    "review_id",
+  ];
+  if (!validSortQueries.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "Invalid sort query!" });
+  }
```
Comment (leekli) [inline api/app.models.js]: You're not required to implement this yet

File: __tests__/api.test.js [hunk 5]
```diff
@@ -59,4 +59,45 @@ describe("Returns with the correct JSON endpoint", () => {
...
+          const expectedOutput = Object.keys(response.body);
+          expect(input).toMatchObject(expectedOutput);
+        });
+      });
+  });
 });
```
Comment (leekli) [inline __tests__/api.test.js:103]: There is 1 test missing, which is to check that `the articles should be sorted by date in descending order.` as the Trello ticket says.

You'll need to add a test for this. The 'jest-sorted' package will help you with this

File: api/app.models.js [hunk 6]
```diff
@@ -6,3 +6,28 @@ exports.getCategory = () => {
...
+  return db
+    .query(
+      "SELECT r.review_id, r.title, r.category, r.designer, r.owner, r.review_img_url, r.created_at, r.votes, COUNT(c.comment_id) AS comment_count FROM reviews AS r LEFT JOIN comments AS c ON r.review_id = c.review_id GROUP BY r.review_id ORDER BY created_at DESC"
```
Comment (leekli) [inline api/app.models.js:19]: Nice sql query, good to see you've used the COUNT aggregate function to help you count the comments, well done!

**Review by leekli — CHANGES_REQUESTED** [review]
Well done so far, Charlie!

Got a couple of change requests which are outlined in my comments, the 2 main ones are:
* adding a .length check to the test
* you need a test to check that the reviews are sorted by date in descending order - see my comments on this

And any other minor changes.

Please review these, make the necessary changes and then submit a new pull request for review.

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**